### PR TITLE
Bring branch to parity with Blue Scarab fixes and address warnings

### PR DESCRIPTION
**Pull request by kirre-bylund**
**Blue Scarab Pull Requests**
https://github.com/LootLocker/unreal-sdk/pull/12
https://github.com/LootLocker/unreal-sdk/pull/9
https://github.com/LootLocker/unreal-sdk/pull/8

I've gone through them commit by commit and added all the fixes that had not been addressed on the v2 branch. The only thing I have not brought in is the Guest Login addition as that is part of a larger discussion on how to handle platforms vs authentication methods in the sdks. In PR 8 see commits 
bac02afa890a4141c22322a798fbe99f742108b5 - sync project to plugin
e51468687b5c76a2b41da5dc834748df6c3a37a1 - Guest Login

** Template **
In June the file LootLockerTemplates.h was added to the public facing part of the v2 branch but never updated nor used anywhere. Until I know why and what it's for I have removed it.

**Compilation Warnings**
I went through and fixed a bunch of warnings in the project:
* "Uninitialized" - not initializing stuff in C++ is bad. So I fixed a slew of those problems where I found them.
* "Auto doesn't catch references, unintended copy" - the auto keyword just deduces the underlying type and if not used as "auto&" it will make a copy of the object which is stupid for for loops for example.
* "Intended bitwise comparison?" - We're checking if the map is empty and the NOT was in the wrong place causing this warning. I wanted to switch it out to an "IsEmpty" but that wasn't available in earlier C# versions.

### CODE REVIEW

File: v2/LootLockerSDK/Source/LootLockerSDK/Private/GameAPI/LootLockerHeroRequestHandler.cpp [hunk 1]
```diff
@@ -66,17 +66,11 @@ void ULootLockerHeroRequestHandler::GetOtherPlayersHeroLoadout(const int32 HeroI
...
+	LLAPI<FLootLockerHeroLoadoutResponse>::CallAPI(HttpClient, FLootLockerHeroAssetInstance{ AssetInstanceID }, ULootLockerGameEndpoints::AddAssetToHeroLoadout, { HeroID }, EmptyQueryParams, OnCompleteBP, OnComplete);
 }
 
 void ULootLockerHeroRequestHandler::RemoveAssetToHeroLoadout(const int32 HeroID, const int32 AssetInstanceID, const FHeroLoadoutReseponseBP &OnCompleteBP, const FHeroLoadoutReseponseDelegate &OnComplete)
 {
-	FLootLockerHeroAssetInstance Asset;
-	Asset.instance_id = AssetInstanceID;
-	
-	LLAPI<FLootLockerHeroLoadoutResponse>::CallAPI(HttpClient, Asset, ULootLockerGameEndpoints::RemoveAssetToHeroLoadout,  { HeroID },EmptyQueryParams,OnCompleteBP, OnComplete);
+	LLAPI<FLootLockerHeroLoadoutResponse>::CallAPI(HttpClient, FLootLockerHeroAssetInstance{ AssetInstanceID }, ULootLockerGameEndpoints::RemoveAssetToHeroLoadout,  { HeroID },EmptyQueryParams,OnCompleteBP, OnComplete);
```
Comment (TobiasBerg) [inline v2/LootLockerSDK/Source/LootLockerSDK/Private/GameAPI/LootLockerHeroRequestHandler.cpp:74]: There's some weird spacing for parameters here

File: v2/LootLockerSDK/Source/LootLockerSDK/Public/LootLockerConfig.h [hunk 2]
```diff
@@ -52,10 +54,10 @@ enum class ELootLockerPlatformType : uint8
 	Ios = 1					UMETA(DisplayName = "Ios"),
 	Steam = 2				UMETA(DisplayName = "Steam"),
 	NintendoSwitch = 3		UMETA(DisplayName = "NintendoSwitch"),
-	EpicStore = 3			UMETA(DisplayName = "EpicStore"),
-	PlayStationNetwork = 3  UMETA(DisplayName = "PlayStationNetwork"),
-	Xbox = 3				UMETA(DisplayName = "Xbox"),
-	GoG = 3					UMETA(DisplayName = "GoG")
```
Comment (TobiasBerg) [inline v2/LootLockerSDK/Source/LootLockerSDK/Public/LootLockerConfig.h:58]: I would remove GoG completely

File: v2/LootLockerSDK/Source/LootLockerSDK/Public/LootLockerConfig.h [hunk 3]
```diff
@@ -52,10 +54,10 @@ enum class ELootLockerPlatformType : uint8
 	Ios = 1					UMETA(DisplayName = "Ios"),
 	Steam = 2				UMETA(DisplayName = "Steam"),
 	NintendoSwitch = 3		UMETA(DisplayName = "NintendoSwitch"),
-	EpicStore = 3			UMETA(DisplayName = "EpicStore"),
```
Comment (TobiasBerg) [inline v2/LootLockerSDK/Source/LootLockerSDK/Public/LootLockerConfig.h:55]: I would remove EpicStore completely

**Review by TobiasBerg — CHANGES_REQUESTED** [review]

File: v2/LootLockerSDK/Source/LootLockerSDK/Public/LootLockerConfig.h (hunk 3 above)
Comment (kirre-bylund) [reply v2/LootLockerSDK/Source/LootLockerSDK/Public/LootLockerConfig.h:55]: Fixed

File: v2/LootLockerSDK/Source/LootLockerSDK/Public/LootLockerConfig.h (hunk 2 above)
Comment (kirre-bylund) [reply v2/LootLockerSDK/Source/LootLockerSDK/Public/LootLockerConfig.h:58]: Fixed

File: v2/LootLockerSDK/Source/LootLockerSDK/Private/GameAPI/LootLockerHeroRequestHandler.cpp (hunk 1 above)
Comment (kirre-bylund) [reply v2/LootLockerSDK/Source/LootLockerSDK/Private/GameAPI/LootLockerHeroRequestHandler.cpp]: Fixed

**Review by JohannesLoot — COMMENTED** [review]
Do we need to test any of the changes from Blue Scarab before merging this?

File: v2/LootLockerSDK/Source/LootLockerSDK/Private/Utils/LootLockerUtilities.h [hunk 4]
```diff
@@ -103,9 +104,9 @@ struct LLAPI
                 Delimiter = "&";
             }
         }
-        UE_LOG(LogTemp, Warning, TEXT("Request:"));
-        UE_LOG(LogTemp, Warning, TEXT("ContentString:%s"), *ContentString);
-        UE_LOG(LogTemp, Warning, TEXT("EndpointWithArguments:%s"), *EndpointWithArguments);
+        UE_LOG(LogLootLockerGameSDK, Log, TEXT("Request:"));
```
Comment (kirre-bylund) [inline v2/LootLockerSDK/Source/LootLockerSDK/Private/Utils/LootLockerUtilities.h:107]: Should still log these on error as warn.

**Comment by kirre-bylund** [comment]
> Do we need to test any of the changes from Blue S

The answer to "Do we need to test the changes" should always be yes. But it's a bit tricky as it's mostly just warning fixes all over the place so what you'd need is a full regression test. Would have been satisfied with CI tests if we had any. So under the circumstances I'd say no. If we still want to do any then I'd say; do some spot checks on some flows, and then test the missions one as that has the most significant changes?

**Review by TobiasBerg — APPROVED** [review]

**Pull request merged**